src/core/defaultappearance.js

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isName, Name } from "./primitives.js";
+import { escapePDFName } from "./core_utils.js";
+
+// Parse DA to extract font and color informations
+function parseDefaultAppearance(str) {
```

Comment:
Given just how *simplistic* this algorithm is, compared to the existing `PartialEvaluator`/`Parser`/`Lexer`, isn't there a very real risk that it won't cope with all the kinds of weirdness seem in real-world PDF document!?

Can't you *at least* re-use the `Parser`/`Lexer` instead here, rather than "re-inventing the wheel" so to speak, since I believe that should be slightly safer (and with less risk of future bugs) overall?

Reply:
Thinking about this more, you should probably just use the [`EvaluatorPreprocessor`](https://github.com/mozilla/pdf.js/blob/f224a0eb9f5cb4798d9505f8b618934e12b19885/src/core/evaluator.js#L3856-L4127) for this parsing as well. That way you'd use the *same* exact parsing as the "regular" evaluator uses, and thus benefit from all of the existing handling of various edge-cases (you'd simply need to `export` it such that it can be used from this file as well).